From 56ba558ac343476a19a360fe139a6896bfc0b9e3 Mon Sep 17 00:00:00 2001
From: Andy Green <andy@openmoko.com>
Date: Tue, 13 May 2008 18:53:39 +0100
Subject: [PATCH] add-force-backlight-up-on-resume.patch

The backlight status is restored during resume action, if it was
off then it resumes like that.

This patch forces it on after resume.

Signed-off-by: Andy Green <andy@openmoko.com>
---
 drivers/i2c/chips/pcf50633.c |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/drivers/i2c/chips/pcf50633.c b/drivers/i2c/chips/pcf50633.c
index b7b4921..dc4ea0a 100644
--- a/drivers/i2c/chips/pcf50633.c
+++ b/drivers/i2c/chips/pcf50633.c
@@ -1952,8 +1952,9 @@ static int pcf50633_suspend(struct device *dev, pm_message_t state)
 
 void pcf50633_backlight_resume(struct pcf50633_data *pcf)
 {
-	__reg_write(pcf, PCF50633_REG_LEDOUT, pcf->standby_regs.ledout);
-	__reg_write(pcf, PCF50633_REG_LEDENA, pcf->standby_regs.ledena);
+	/* we force the backlight on in fact */
+	__reg_write(pcf, PCF50633_REG_LEDOUT, 0x3f);
+	__reg_write(pcf, PCF50633_REG_LEDENA, 0x21);
 	__reg_write(pcf, PCF50633_REG_LEDDIM, pcf->standby_regs.leddim);
 }
 EXPORT_SYMBOL_GPL(pcf50633_backlight_resume);
-- 
1.5.6.5

